skimage/color/colorconv.py

```diff
@@ -852,77 +940,36 @@ def gray2rgba(image, alpha=None):
     if not np.can_cast(alpha, arr.dtype):
         warn("alpha can't be safely cast to image dtype {}"
              .format(arr.dtype.name), stacklevel=2)
-
-    rgba = np.empty(arr.shape + (4, ), dtype=arr.dtype)
-    rgba[..., :3] = arr[..., np.newaxis]
-    rgba[..., 3] = alpha
-
+    if np.isscalar(alpha):
+        alpha = np.full(arr.shape, alpha, dtype=arr.dtype)
+    elif alpha.shape != arr.shape:
+        raise ValueError("alpha.shape must match image.shape")
+    rgba = np.stack((arr,) * 3 + (alpha,), axis=channel_axis)
     return rgba
 
 
-def gray2rgb(image, alpha=None):
+def gray2rgb(image, *, channel_axis=-1):
```

Comment:
The `alpha` argument may had be removed in a different PR, but it may be more complicated. Please remove the corresponding item in TODO.txt :wink:

Reply:
Apparently I did already remove `alpha` and the TODO item as part of #5463 so perhaps we should merge that first to make this cleaner.